### DISCUSSION TITLE
how to use Koa-multer and Koa-router as service object in feathersjs?

### DISCUSSION THREAD

**Discussion by SYBIOTE** [feathersjs/feathers#3184] [Q&A] [answered]
migrating an app from feathersjs/express to feathersjs/koa
feathersjs/express uses Multer and you can use it in feathers with this syntax 

`app.use("path",multipartmiddleware.any(),middleware1,middleware2) // middleware 2 can be a service definition `

by service definition I mean this 

`{
  create: function1() // and so on for other functions
  }`

following this I can use 

`const service = app.service("path")`
and ill have the service well defined 

but in feathersjs/koa , you need to use koa-multer right? and you need Koa-router to use it 
Edit: looks like I can use body parser for this, but that's yet to be updated
eg
`router.post("path",multipartmiddleware.any(),middleware1,middleware2) // middleware are async functions here`

the second middleware function can directly call function1() so service definition is not needed

I do register the routes following this 
`app.use(router.routes())`

since I need to use koa-router for koa-multer, how do I register the router as a service? or is there another way to do it 

the main thing is that I need to upload a  .glb with a feathersjs/koa application

I saw this discussion 
[https://github.com/feathersjs/feathers/discussions/3097](url)

I hope more can be elaborated on this

**Comment by SYBIOTE** [comment, marked as answer]
i cant seem to find a way to use koa-router , but you can use before: service middle wate with koa multer instead
example snippet

import Multer from 'koa-multer' \
const multipartMiddleware = Multer({ limits: { fieldSize: Infinity, files: 1 } })\
// other code\
app.use(\
    &nbsp;'upload'\,
    &nbsp;{\
    &ensp; create: upload(app)\
    &nbsp;},\
    &nbsp;{\
    &ensp;koa: {\
    &emsp;before: [\
     &emsp; &nbsp;    multipartMiddleware.any(),\
     &emsp;  &nbsp;    async (ctx, next) => {\
     &emsp;   &nbsp;     console.log('trying to upload file')\
    &emsp;   &nbsp;      // handle as needed\
       &emsp;    &nbsp;  await next()\
       &emsp;   &nbsp;   console.log('uploaded file')\
      &emsp;    &nbsp;   return ctx.body\
     &emsp;     }\
      &ensp;   ]\
   &nbsp;   }\
    }

**Reply by vicentecastroa** [reply]
Thanks!
